### DISCUSSION TITLE
[Support]: Frigate is not publishing to "frigate/events" when camera "Alerts"

### DISCUSSION THREAD

**Discussion by Handrail9** [blakeblackshear/frigate#13880] [General Support] [answered]
### Describe the problem you are having

Hello, this is a continuation of [Issue #281](https://github.com/SgtBatten/HA_blueprints/issues/281) of SgtBattens Frigate Notification Blueprint. The issue has come down to Frigate not pushing anything to MQTT when an alert happens. 

### Version

0.14.1-f4f3cfa

### What browser(s) are you using?

Librewolf 130.0-3

### Frigate config file

```yaml
mqtt:
  enabled: true
  host: 192.168.1.80
  user: USERNAME
  password: PASSWORD

detectors:
  coral:
    type: edgetpu
    device: pci

cameras:
  camera: # <--- this will be changed to your actual camera later
    enabled: true
    ffmpeg:
      inputs:
        - path: rtsp://192.168.1.128:8080/video/h264
          roles:
            - record
            - detect
    detect:
      enabled: true
    zones:
      Zone:
        coordinates: 0.435,0.964,0.445,0.515,0.162,0.515,0.159,0.97
        loitering_time: 20
        objects: person
    motion:
      mask: 0.435,0.966,0.444,0.521,0.164,0.515,0.157,0.967
    record: # <----- Enable recording
      enabled: True
version: 0.14
```

### Relevant Frigate log output

```shell
UI doesn't show any logs, it is blank.
These are the docker compose logs

frigate       | 2024-09-21 22:08:08.058671025  192.168.1.134 - - [21/Sep/2024:22:08:08 -0400] "GET /api/review/summary?timezone=Atlantic%2FReykjavik HTTP/1.1" 200 1634 "http://192.168.1.80:5000/review" "Mozilla/5.0 (Windows NT 10.0; rv:130.0) Gecko/20100101 Firefox/130.0" "-"
frigate       | 2024-09-21 22:08:08.058703379  192.168.1.134 - - [21/Sep/2024:22:08:08 -0400] "GET /api/config HTTP/1.1" 200 7849 "http://192.168.1.80:5000/review" "Mozilla/5.0 (Windows NT 10.0; rv:130.0) Gecko/20100101 Firefox/130.0" "-"
frigate       | 2024-09-21 22:08:37.401937984  [2024-09-21 22:08:37] frigate.comms.mqtt             ERROR   : Unable to publish to stats: client is not connected
frigate       | 2024-09-21 22:09:02.366202875  127.0.0.1 - - [21/Sep/2024:22:09:02 -0400] "" 400 0 "-" "-" "-"
```

### Relevant go2rtc log output

```shell
info
2024-09-21 09:14:53
startup
Preparing new go2rtc config...
info
2024-09-21 09:14:54
startup
Starting go2rtc...
info
2024-09-21 09:14:54
startup
go2rtc platform=linux/amd64 revision=b2399f3 version=1.9.2
info
2024-09-21 09:14:54
startup
config path=/dev/shm/go2rtc.yaml
info
2024-09-21 09:14:54
rtsp
listen addr=:8554
info
2024-09-21 09:14:54
api
listen addr=:1984
info
2024-09-21 09:14:54
webrtc
listen addr=:8555/tcp
info
2024-09-21 09:15:03
startup
Starting go2rtc healthcheck service...
```

### FFprobe output from your camera

```shell
/opt/frigate# ffprobe rtsp://192.168.1.128:8080/video/h264
ffprobe version n5.1-2-g915ef932a3-20220731 Copyright (c) 2007-2022 the FFmpeg developers
  built with gcc 12.1.0 (crosstool-NG 1.25.0.55_3defb7b)
  configuration: --prefix=/ffbuild/prefix --pkg-config-flags=--static --pkg-config=pkg-config --cross-prefix=x86_64-ffbuild-linux-gnu- --arch=x86_64 --target-os=linux --enable-gpl --enable-version3 --disable-debug --enable-iconv --enable-libxml2 --enable-zlib --enable-libfreetype --enable-libfribidi --enable-gmp --enable-lzma --enable-fontconfig --enable-libvorbis --enable-opencl --enable-libpulse --enable-libvmaf --enable-libxcb --enable-xlib --enable-amf --enable-libaom --enable-libaribb24 --enable-avisynth --enable-libdav1d --enable-libdavs2 --disable-libfdk-aac --enable-ffnvcodec --enable-cuda-llvm --enable-frei0r --enable-libgme --enable-libass --enable-libbluray --enable-libjxl --enable-libmp3lame --enable-libopus --enable-mbedtls --enable-librist --enable-libtheora --enable-libvpx --enable-libwebp --enable-lv2 --enable-libmfx --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-librav1e --enable-librubberband --disable-schannel --enable-sdl2 --enable-libsoxr --enable-libsrt --enable-libsvtav1 --enable-libtwolame --enable-libuavs3d --enable-libdrm --enable-vaapi --enable-libvidstab --enable-vulkan --enable-libshaderc --enable-libplacebo --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libzimg --enable-libzvbi --extra-cflags=-DLIBTWOLAME_STATIC --extra-cxxflags= --extra-ldflags=-pthread --extra-ldexeflags=-pie --extra-libs='-ldl -lgomp' --extra-version=20220731
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
[h264 @ 0x55dd426a4c40] idr_pic_id is invalid
    Last message repeated 2 times
Input #0, rtsp, from 'rtsp://192.168.1.128:8080/video/h264':
  Duration: N/A, start: -0.658188, bitrate: N/A
  Stream #0:0: Video: h264 (Baseline), yuv420p(tv, unknown/bt470bg/unknown, progressive), 1280x720, 24.33 tbr, 90k tbn
  Stream #0:1: Audio: aac (LC), 48000 Hz, mono, fltp
```

### Frigate stats

```json
{"cameras":{"camera":{"audio_dBFS":0.0,"audio_rms":0.0,"camera_fps":5.0,"capture_pid":474,"detection_enabled":true,"detection_fps":0.0,"ffmpeg_pid":489,"pid":467,"process_fps":5.0,"skipped_fps":0.0}},"cpu_usages":{"1":{"cmdline":"/package/admin/s6/command/s6-svscan -d4 -- /run/service","cpu":"0.0","cpu_average":"0","mem":"0.0"},"106":{"cmdline":"bash ./run.user go2rtc-healthcheck","cpu":"0.0","cpu_average":"0","mem":"0.0"},"107":{"cmdline":"python3 -u -m frigate","cpu":"0.4","cpu_average":"0","mem":"0.3"},"125":{"cmdline":"nginx: master process nginx","cpu":"0.0","cpu_average":"0","mem":"0.0"},"129920":{"cmdline":"bash","cpu":"0.0","cpu_average":"0","mem":"0.0"},"130484":{"cmdline":"sleep 60","cpu":"0.0","cpu_average":"0","mem":"0.0"},"130593":{"cmdline":"sleep 30s","cpu":"0.0","cpu_average":"0","mem":"0.0"},"15":{"cmdline":"s6-supervise s6-linux-init-shutdownd","cpu":"0.0","cpu_average":"0","mem":"0.0"},"158":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"159":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"16":{"cmdline":"/package/admin/s6-linux-init/command/s6-linux-init-shutdownd -c /run/s6/basedir -g 3000 -C -B","cpu":"0.0","cpu_average":"0","mem":"0.0"},"160":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"161":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"162":{"cmdline":"nginx: cache manager process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"24":{"cmdline":"s6-supervise s6rc-oneshot-runner","cpu":"0.0","cpu_average":"0","mem":"0.0"},"25":{"cmdline":"s6-supervise s6rc-fdholder","cpu":"0.0","cpu_average":"0","mem":"0.0"},"26":{"cmdline":"s6-supervise certsync","cpu":"0.0","cpu_average":"0","mem":"0.0"},"27":{"cmdline":"s6-supervise certsync-log","cpu":"0.0","cpu_average":"0","mem":"0.0"},"28":{"cmdline":"s6-supervise frigate","cpu":"0.0","cpu_average":"0","mem":"0.0"},"29":{"cmdline":"s6-supervise frigate-log","cpu":"0.0","cpu_average":"0","mem":"0.0"},"296":{"cmdline":"bash ./run.user certsync","cpu":"0.0","cpu_average":"0","mem":"0.0"},"30":{"cmdline":"s6-supervise go2rtc","cpu":"0.0","cpu_average":"0","mem":"0.0"},"31":{"cmdline":"s6-supervise go2rtc-healthcheck","cpu":"0.0","cpu_average":"0","mem":"0.0"},"32":{"cmdline":"s6-supervise go2rtc-log","cpu":"0.0","cpu_average":"0","mem":"0.0"},"33":{"cmdline":"s6-supervise nginx","cpu":"0.0","cpu_average":"0","mem":"0.0"},"34":{"cmdline":"s6-supervise nginx-log","cpu":"0.0","cpu_average":"0","mem":"0.0"},"397":{"cmdline":"frigate.logger       ","cpu":"0.0","cpu_average":"0","mem":"0.1"},"421":{"cmdline":"frigate.recording_manager","cpu":"0.3","cpu_average":"0","mem":"0.2"},"430":{"cmdline":"frigate.review_segment_manager","cpu":"0.1","cpu_average":"0","mem":"0.1"},"449":{"cmdline":"/usr/bin/python3 -c from multiprocessing.resource_tracker import main;main(56)","cpu":"0.0","cpu_average":"0","mem":"0.0"},"45":{"cmdline":"/package/admin/s6/command/s6-ipcserverd -1 -- /package/admin/s6/command/s6-ipcserver-access -v0 -E -l0 -i data/rules -- /package/admin/s6/command/s6-sudod -t 30000 -- /package/admin/s6-rc/command/s6-rc-oneshot-run -l ../.. --","cpu":"0.0","cpu_average":"0","mem":"0.0"},"450":{"cmdline":"frigate.detector.coral","cpu":"0.0","cpu_average":"0","mem":"0.2"},"452":{"cmdline":"frigate.output       ","cpu":"0.2","cpu_average":"0","mem":"0.1"},"46":{"cmdline":"/package/admin/s6-2.11.3.2/command/s6-fdholderd -1 -i data/rules","cpu":"0.0","cpu_average":"0","mem":"0.0"},"465":{"cmdline":"ffmpeg -threads 1 -f rawvideo -pix_fmt yuv420p -video_size 1280x720 -i pipe: -threads 1 -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:","cpu":"0.0","cpu_average":"0","mem":"0.0"},"467":{"cmdline":"frigate.process:camera","cpu":"0.6","cpu_average":"4","mem":"0.2"},"474":{"cmdline":"frigate.capture:camera","cpu":"0.6","cpu_average":"0","mem":"0.1"},"476":{"cmdline":"ffmpeg -threads 1 -f rawvideo -pix_fmt yuv420p -video_size 1280x720 -i pipe: -threads 1 -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:","cpu":"0.0","cpu_average":"0","mem":"0.0"},"489":{"cmdline":"ffmpeg -hide_banner -loglevel warning -threads 2 -user_agent FFmpeg Frigate/0.14.1-f4f3cfa -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://192.168.1.128:8080/video/h264 -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an /tmp/cache/camera@%Y%m%d%H%M%S%z.mp4 -r 5 -vf fps=5,scale=1280:720 -threads 2 -f rawvideo -pix_fmt yuv420p pipe:","cpu":"3.3","cpu_average":"2","mem":"0.0"},"7003":{"cmdline":"/usr/bin/python3 -c from multiprocessing.resource_tracker import main;main(46)","cpu":"0.0","cpu_average":"0","mem":"0.0"},"87":{"cmdline":"s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/nginx","cpu":"0.0","cpu_average":"0","mem":"0.0"},"88":{"cmdline":"s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/go2rtc","cpu":"0.0","cpu_average":"0","mem":"0.0"},"89":{"cmdline":"s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/certsync","cpu":"0.0","cpu_average":"0","mem":"0.0"},"90":{"cmdline":"s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/frigate","cpu":"0.0","cpu_average":"0","mem":"0.0"},"98":{"cmdline":"/usr/local/go2rtc/bin/go2rtc -config=/dev/shm/go2rtc.yaml","cpu":"0.0","cpu_average":"0","mem":"0.0"},"frigate.full_system":{"cpu":"2.2","mem":"33.5"}},"detection_fps":0.0,"detectors":{"coral":{"detection_start":0.0,"inference_speed":6.13,"pid":450}},"processes":{"go2rtc":{"pid":98},"logger":{"pid":397},"recording":{"pid":421},"review_segment":{"pid":430}},"service":{"last_updated":1726971232,"latest_version":"0.14.1","storage":{"/dev/shm":{"free":509.2,"mount_type":"tmpfs","total":512.0,"used":2.8},"/media/frigate/clips":{"free":8963981.6,"mount_type":"xfs","total":15257610.0,"used":6293628.4},"/media/frigate/recordings":{"free":8963981.6,"mount_type":"xfs","total":15257610.0,"used":6293628.4},"/tmp/cache":{"free":4763.2,"mount_type":"tmpfs","total":4768.4,"used":5.2}},"temperatures":{"apex_0":40.8},"uptime":46736,"version":"0.14.1-f4f3cfa"}}
```

### Operating system

Other Linux

### Install method

Docker Compose

### Object Detector

Coral

### Network connection

Wired

### Camera make and model

Unsure. Is an android phone running an IP Cam software from F-Droid.

### Screenshots of the Frigate UI's System metrics pages

![image](https://github.com/user-attachments/assets/e04000f7-86d8-4fef-a5f8-a5530848098f)

### Any other information that may be helpful

I'm a little stupid so ELI5 if possible :+1: 
I see that Frigate is having some sort of issue publishing to MQTT in the log, but I can see quite a few things being published by Frigate at the same time I was trying to test. I am able to see the alert in the Frigate UI, but I am not seeing anything in mqtt-explorer that events was published to. I've attached a screenshot below of what I am able to see in mqtt-explorer a few moments after triggering an alert by walking in front of my camera.
![image](https://github.com/user-attachments/assets/1ca9c458-e91a-43fb-9f02-e7bf31bc9258)

**Comment by SgtBatten** [comment]
specifically the lack of events topic being posted. You are clearly getting frigate topics, but apparently not events.

**Comment by SgtBatten** [comment]
> ```shell
> ERROR   : Unable to publish to stats: client is not connected
> ```

noted this. What port is your mqtt container using?

**Reply by Handrail9** [reply]
I exposed port 8093
```
    ports:
      - 8093:1883
```

Does the port make a difference? I'll try testing on the original port

**Reply by NickM-27** [reply, marked as answer]
you would have to specify that port in frigate's config then

**Reply by Handrail9** [reply]
Oh jeez. It's always something stupid and small i overlooked when I have issues. Thank you both :sob:
